wordpress-mobile/WordPress-Android
# Replace GravatarUtils from WPUtils, use Gravatar-SDK-Android instead

## What changed
Replace GravatarUtils from WPUtils, use Gravatar-SDK-Android instead

- Renamed `fixGravatarUrl` to `rewriteAvatarUrl`, this function applies on gravatar url and any image url. It removes the query params and update it to force specific size and default avatar parameters. Ideally I wanted to reuse the `GravatarUtils` from Gravatar-SDK-Android, but we're not sure exactly what kind of url we'll get from the wordpress backend.
- Renamed the file that contained `fixGravatarUrl` from `GravatarUtils` to `WPAvatarUtils`, also renamed `GravatarUtilsWrapper` to `WPAvatarUtilsWrapper`.
- Removed all usage to `WordPress-Utils-Android.GravatarUtils`, use function `Gravatar-SDK-Android.GravatarUtils` instead.
- Removed `blavatarFromUrl` functions, they were not used and they are specific to wordpress.com sites.
- Added some documentation on `WPAvatarUtils` to make things clearer.
- Used the same default image (Mystery Man, that was renamed to Mystery Person).

⚠️ Things to double check ⚠️ 
- Defaults are the same as in [GravatarUtils.java](https://github.com/wordpress-mobile/WordPress-Utils-Android/blob/trunk/WordPressUtils/src/main/java/org/wordpress/android/util/GravatarUtils.java)

## How to test

1. Use a `jetpack` variant (eg. `jetpackWasabiDebug`)
2. Test the Site Comments screen: My Site tab -> More -> Comments, make sure avatars are loading
3. Test the Reader Comments screen: Reader tab -> Discover -> tap a post -> scroll down -> tap "View all comments", make sure avatars are loading
4. Test the Notifications: Notifications tab -> scroll down -> make sure avatars are loading
5. Test the Me/Upload: Me tab -> Tap avatar -> Change avatar -> make sure avatar is updated
